ocaml-lsp-server/src/signature_help.ml

```diff
@@ -195,51 +42,24 @@ let run (state : State.t) { SignatureHelpParams.textDocument = { uri }; position
     let store = state.store in
     Document_store.get store uri
   in
-  let pos = Position.logical position in
-  let prefix =
-    (* The value of [short_path] doesn't make a difference to the final result
-       because labels cannot include dots. However, a true value is slightly
-       faster for getting the prefix. *)
-    Compl.prefix_of_position (Document.source doc) pos ~short_path:true
-  in
-  (* TODO use merlin resources efficiently and do everything in 1 thread *)
   match Document.kind doc with
-  | `Other ->
-    let help = SignatureHelp.create ~signatures:[] () in
-    Fiber.return help
+  | `Other -> Fiber.return (SignatureHelp.create ~signatures:[] ())
   | `Merlin merlin ->
-    let* application_signature =
-      let* inside_comment = Check_for_comments.position_in_comment ~position ~merlin in
-      match inside_comment with
-      | true -> Fiber.return None
-      | false ->
-        Document.Merlin.with_pipeline_exn ~name:"signature-help" merlin (fun pipeline ->
-          let typer = Mpipeline.typer_result pipeline in
-          let pos = Mpipeline.get_lexing_pos pipeline pos in
-          let node = Mtyper.node_at typer pos in
-          application_signature node ~prefix)
```

Comment:
I thin you should be able to use `Merlin_analysis.Signature_help.application_signature` as a drop-in replacement here and keep the rest of the `run` function identical :-)